packages/parchment/lib/src/convert/html.dart

```diff
@@ -321,8 +328,10 @@ class _ParchmentHtmlEncoder extends Converter<Delta, String> {
       position = buffer.length;
     }
 
-    assert(openBlockTags.length <= 1,
-        'At most one paragraph should be pushed in stack');
+    // TODO This assert currently fails the 'Multi-level lists with trailing paragraph' test.
```

Comment:
Shouldn't we try to fix this?

Reply:
We should and I attempted to, but I don't quite understand how the code is handling it. I think we should probably close a block if the indent level changes, or the block type. I could take another look at this today. Any hints you may have would be appreciated.

Reply:
prior to handling the plain block, we handle any previously opened block. In this instance the stack is not being emptied - only the last inserted element (list with indent = 1) was popped, the list element with no indent remains.
Later, the paragraph block is added to the stack (thus more than one)

Reply:
I'm working on this...

Reply:
This is fixed. It took quite a bit of time to figure out the proper way to close blocks that would work for everything.